plat/rpi/rpi5/scmi/scmi_policy.c

```diff
@@ -12,18 +12,77 @@
 
 #include "rpi5_scmi_resources.h"
 
+_Static_assert(sizeof(scmi_umask_t) * 8 > SCMI_NUM_AGENTS);
+
+static const struct scmi_device default_dev = {
+		.rsts = (int[]){-1},
+		.pins = (int[]){-1},
+};
+
+static const struct scmi_device rpi_scmi_devices[RPI5_SCMI_DEV_COUNT] = {
```

Comment:
This is quite complex logic and I see no benefit in it. Instead of mapping devices to resets, you can map resets to devices. E.g. create simple array "reset_to_dev = [0, RPI5_SCMI_DEV_PCIE1, RPI5_SCMI_DEV_PCIE2, 0 , 0], etc. It will be much easier any simpler.

Reply:
First, this logic follows approach:
- already used by xen-troops in other projects;
- which follows DT definitions.

Second, I can't agree that it will be simpler - lets imagine we have not 4, but 64 resets. It will be total pain trying to fill device table with approach you've proposed. With current approach - all that is required is to follow DT.